src/v2/modules/addresses/controller.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const addressService = require('../../services/address');
+
+exports.postAddress = async (request, h) => {
+  const addressData = request.payload;
+  const { error, address } = await addressService.createAddress(addressData);
+
+  return error
+    ? h.response(error).code(422)
```

Comment:
This could use mapErrorResponse

Reply:
I did try that but Hapi does not return the message body that I was hoping for. 

Because this is a private API, it feels like it would be good to reflect the validation errors that the service layer returns. But when the response is a Boom error the message body was not satisfactory, it could of course still be mapped, but perhaps not to a Boom error in this case.

Reply:
It turns out Boom/Hapi wants to protect devs from themselves so only output a small 'safe' portion of the error to avoid any unexpected leakage. Instead the Boom error is now created, then augmented with the payload that will be OK to transmit on this private API.